storagemarket/provider_loop.go

```diff
@@ -72,11 +72,22 @@ func (p *Provider) loop() {
 				continue
 			}
 
+			// Tag the storage required for the deal in the staging area
+			err = p.storageManager.Tag(p.ctx, deal.DealUuid, deal.ClientDealProposal.Proposal.PieceSize)
+			if err != nil {
+				go writeDealResp(false, &api.ProviderDealRejectionInfo{
```

Comment:
Do we need to untag funds here as well?

Reply:
Yes, we do. I am not happy with the current pattern, but maybe this won't get more complex so is good enough for now - https://github.com/filecoin-project/boost/pull/103/commits/bd7616124504bbfca6b34870e2f9c0484ad9a330